osm2streets/src/render.rs

```diff
@@ -252,6 +261,45 @@ impl StreetNetwork {
         let output = serde_json::to_string_pretty(&obj)?;
         Ok(output)
     }
+
+    /// For an intersection, show all the movements.
+    pub fn debug_movements_geojson(&self, timer: &mut Timer) -> Result<String> {
+        let initial_map = crate::initial::InitialMap::new(self, timer);
+
+        let mut pairs = Vec::new();
+
+        for (i, intersection) in &self.intersections {
+            // Find the endpoints
+            let road_points: Vec<_> = intersection
+                .roads
+                .iter()
+                .map(|r| {
+                    let pl = &initial_map.roads[r].trimmed_center_pts;
+                    let pt = if r.i1 == *i {
+                        pl.first_pt()
+                    } else {
+                        pl.last_pt()
+                    };
+                    pt
+                })
+                .collect();
+            for (a, b) in intersection.movements.iter() {
+                if *a != *b {
+                    pairs.push((
+                        Line::must_new(road_points[*a], road_points[*b])
```

Comment:
Oh, this can fail if the points are too close together... I find it surprising that a line type can't even represent lines segments of length zero... I guess that would mean that it has no direction, and so a whole host of operations on it would need to be fallible...

I'll use the constructor that returns a result instead.